oxd-server/src/main/java/org/gluu/oxd/server/op/RegisterSiteOperation.java

```diff
@@ -420,18 +422,26 @@ private void validateParametersAndFallbackIfNeeded(RegisterSiteParams params) {
     private void persistRp(String siteId, RegisterSiteParams params) {
 
         try {
-            rp = createRp(siteId, params);
+            final RegisterRequest registerRequest = createRegisterClientRequest(params, siteId);
+            rp = createRp(registerRequest);
+            rp.setOxdId(siteId);
+            rp.setApplicationType("web");
+            rp.setOpHost(params.getOpHost());
+            rp.setOpDiscoveryPath(params.getOpDiscoveryPath());
+            rp.setUiLocales(params.getUiLocales());
 
             if (!hasClient(params)) {
-                final RegisterResponse registerResponse = registerClient(params);
+                LOG.info("Save RegisterRequest object to register in OP ...");
+                final RegisterResponse registerResponse = registerClient(params, registerRequest);
+
                 rp.setClientId(registerResponse.getClientId());
                 rp.setClientSecret(registerResponse.getClientSecret());
                 rp.setClientRegistrationAccessToken(registerResponse.getRegistrationAccessToken());
                 rp.setClientRegistrationClientUri(registerResponse.getRegistrationClientUri());
                 rp.setClientIdIssuedAt(registerResponse.getClientIdIssuedAt());
                 rp.setClientSecretExpiresAt(registerResponse.getClientSecretExpiresAt());
             }
-
+            LOG.info("Saving Relying Party object in oxd ...");
```

Comment:
We don't external log for saving. If you really want such log then it should be inside create method.

Reply:
Changes done